File: Spring Security Tutorials/02nd_Authentication_Database/src/main/java/com/example/second/config/SecurityConfiguration.java
package com.example.second.config;

import com.example.second.services.JPAUserDetailsService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.crypto.password.NoOpPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Configuration
public class SecurityConfiguration {

    @Bean
    // definition of password encoder
    public PasswordEncoder passwordEncoder() {
        // replace with BCryptPasswordEncoder
        return NoOpPasswordEncoder.getInstance();
    }
}

File: Project/IDM/src/main/kotlin/com/project/idm/config/SecurityConfiguration.kt
package com.project.idm.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.SecurityConfigurerAdapter
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.web.DefaultSecurityFilterChain
import org.springframework.security.web.SecurityFilterChain


/**
 * Component that manages user details
 * - where to get credentials froms
 */
@Configuration
@EnableWebSecurity
class SecurityConfiguration : SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity>() {

    @Bean
    @Throws(Exception::class)
    fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {

        return http

            .authorizeHttpRequests()
                .requestMatchers("/verify-token", "/check-username/**").permitAll()
                .requestMatchers("/css/**", "/js/**", "/images/**").permitAll()
                .requestMatchers("/invalidate-token").permitAll()
                .requestMatchers("/register").permitAll()
                .requestMatchers("/**").hasAnyAuthority("read", "write")
                .anyRequest().authenticated()
                .and()

            .formLogin()
                .loginPage("/login")
                .defaultSuccessUrl("/")
                .permitAll()
                .and()

            .logout()
                .logoutSuccessUrl("/login")
                .permitAll()
                .and()

            .build()
    }

    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }

}